### ISSUE TITLE
Multiline strings comparison

### ISSUE THREAD

**Issue by vsuharnikov** [softwaremill/diffx#69]
When you compare multiline strings, diffx says that they are different (obviously). But the output is hard to read.

Sample test:
```scala
import com.softwaremill.diffx.scalatest.DiffMatcher
import org.scalatest.freespec.AnyFreeSpec
import org.scalatest.matchers.should.Matchers

class StringComparisonTestSuite extends AnyFreeSpec with Matchers with DiffMatcher {
  "multiline strings" in {
    val s1 = """a1
b2
c3"""

    val s2 = """a1
d2
c3"""

    s1 should matchTo(s2)
  }
}
```

will print:
```
a1
b2
c3 -> a1
d2
c3
```
(red before "->" and green after).

It would be great to have other output, e.g.:
```
a1
b2 -> d2
c3
```

The simplest solution IMO is to split strings by lines and compare them as lists.

What do you think?

**Comment by ghostbuster91** [comment]
Hi, 

This seems like a cool idea, although I am not sure if that should be in the library. I will have to think about it. In the mean time you can tweek diffx by yourself to achieve that.

Here is the code:
```
  implicit val sd: Diff[String] = diffForIterable[String, List](diffForOption(implicitly[Derived[Diff[String]]]))
    .contramap[String](_.split("\n").toList)
```
It is a little bit more complicated that it thought it would be, because we are relaying on `Diff[String] ` inside our definition. I will also think if that could be done simpler. Maybe the design of implicits is wrong.

**Comment by ghostbuster91** [comment]
Native support for multiline strings released in 0.3.18 :)
